cmd/clusterctl/client/cluster/installer.go

```diff
@@ -74,9 +82,50 @@ func (i *providerInstaller) Install() ([]repository.Components, error) {
 
 		ret = append(ret, components)
 	}
+
+	// Verify that Deployments in any components to be installed is available
+	// before moving on.
+	for _, components := range i.installQueue {
+		for _, obj := range components.InstanceObjs() {
+			if obj.GetKind() == "Deployment" {
+				err := i.checkDeploymentIsAvailable(obj)
+				if err != nil {
+					return nil, err
+				}
+			}
+		}
+	}
 	return ret, nil
 }
 
+func (i *providerInstaller) checkDeploymentIsAvailable(obj unstructured.Unstructured) error {
+	c, err := i.proxy.NewClient()
```

Comment:
Would it make sense to move this client instantiation outside of the loop?

Reply:
Whoops...good catch. I'll refactor this.